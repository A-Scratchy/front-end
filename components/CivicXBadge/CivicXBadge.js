import React from 'react';
import PropTypes from 'prop-types';
import styles from './CivicXBadge.css';
import OutboundLink from '../_common_/OutboundLink/OutboundLink';

CivicXBadge.propTypes = {
  sourceUrl: PropTypes.string,
};

CivicXBadge.defaultProps = {
  sourceUrl: 'http://cvcx.org/veterans-solutions-lab/',
};

function CivicXBadge({ sourceUrl }) {
  return (
    <div>
      <OutboundLink
        hasIcon={false}
        href={sourceUrl}
        analyticsEventLabel={`[CivicX Accelerator Badge] ${sourceUrl}`}
      >
        <img
          alt="Operation Code is a graduate of the CivicX Accelerator."
          className={styles.civic}
          src="/static/images/civic-x.png"
        />
      </OutboundLink>
    </div>
  );
}

export default CivicXBadge;
