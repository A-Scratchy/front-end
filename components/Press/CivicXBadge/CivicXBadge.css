@import "../../../common/styles/media-queries.css";

.civic {
  display: block;
  margin: 0 auto;
}

@media (--medium-viewport) {
  .civic {
    width: 100%;
  }
}
