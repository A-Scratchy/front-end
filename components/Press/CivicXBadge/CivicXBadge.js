import React from 'react';
import PropTypes from 'prop-types';
import OutboundLink from 'components/_common_/OutboundLink/OutboundLink';
import styles from './CivicXBadge.css';

CivicXBadge.propTypes = {
  sourceUrl: PropTypes.string,
};

CivicXBadge.defaultProps = {
  sourceUrl: 'http://cvcx.org/veterans-solutions-lab/',
};

function CivicXBadge({ sourceUrl = 'http://cvcx.org/veterans-solutions-lab/' }) {
  return (
    <div>
      <OutboundLink
        href={sourceUrl}
        analyticsEventLabel={`[CivicX Accelerator Badge] ${sourceUrl}`}
      >
        <img
          alt="Operation Code is a graduate of the CivicX Accelerator."
          className={styles.civic}
          src="/static/images/civic-x.png"
        />
      </OutboundLink>
    </div>
  );
}

export default CivicXBadge;
