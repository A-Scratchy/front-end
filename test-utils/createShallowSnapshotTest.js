/* eslint-env jest */
import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers or larger components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
