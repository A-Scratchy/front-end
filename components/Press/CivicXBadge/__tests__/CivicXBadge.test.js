import React from 'react';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';

import CivicXBadge from '../CivicXBadge';

describe('CivicXBadge', () => {
  it('should render with no props passed', () => createShallowSnapshotTest(<CivicXBadge />));
});
