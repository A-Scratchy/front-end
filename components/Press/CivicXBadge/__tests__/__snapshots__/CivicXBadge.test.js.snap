// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CivicXBadge should render with no props passed 1`] = `
<div>
  <OutboundLink
    analyticsEventLabel="[CivicX Accelerator Badge] http://cvcx.org/veterans-solutions-lab/"
    className=""
    hasIcon={false}
    href="http://cvcx.org/veterans-solutions-lab/"
  >
    <img
      alt="Operation Code is a graduate of the CivicX Accelerator."
      className="civic"
      src="/static/images/civic-x.png"
    />
  </OutboundLink>
</div>
`;
